titus-server-runtime/src/main/java/com/netflix/titus/runtime/endpoint/admission/JobRuntimePredictionSelectors.java

```diff
@@ -41,51 +61,146 @@ public static JobRuntimePredictionSelector never() {
+                : Optional.of(JobRuntimePredictionSelection.newBuilder().withPrediction(predictions.getPredictions().last()).withMetadata(allMetadata).build());
     }
 
     /**
-     * Select prediction with the lowest confidence level which is above the provided confidenceThreshold.
+     * Select prediction with the highest confidence if it is below `runtimeThresholdInSeconds` and if the estimated
+     * standard deviation of the predicted distribution is below `sigmaThreshold`
      */
     public static JobRuntimePredictionSelector aboveThreshold(double runtimeThresholdInSeconds,
                                                               double sigmaThreshold,
                                                               Map<String, String> selectionMetadata) {
+        Map<String, String> allMetadata = appendSelectorInfoIfMissing(
+                selectionMetadata,
+                () -> String.format("aboveThreshold{runtimeThresholdInSeconds=%s, sigmaThreshold=%s}, createdAt: %s",
+                        runtimeThresholdInSeconds, sigmaThreshold,
+                        DateTimeExt.toLocalDateTimeString(System.currentTimeMillis())
+                )
+        );
+
         return (jobDescriptor, predictions) -> {
             SortedSet<JobRuntimePrediction> predictionsSet = predictions.getPredictions();
             if (CollectionsExt.isNullOrEmpty(predictionsSet) || predictionsSet.size() < 2) {
                 return Optional.empty();
             }
 
+            JobRuntimePrediction first = predictionsSet.first();
             JobRuntimePrediction last = predictionsSet.last();
+            if (!JobRuntimePredictionUtil.expectedQuantiles(first, last)) {
+                return Optional.empty();
+            }
+
             if (last.getRuntimeInSeconds() > runtimeThresholdInSeconds) {
                 return Optional.empty();
             }
 
-            JobRuntimePrediction first = predictionsSet.first();
             double sigma = (last.getRuntimeInSeconds() - first.getRuntimeInSeconds()) / JobRuntimePredictionUtil.NORM_SIGMA;
             if (sigma > sigmaThreshold) {
                 return Optional.empty();
             }
             return Optional.of(JobRuntimePredictionSelection.newBuilder()
                     .withPrediction(last)
-                    .withMetadata(selectionMetadata)
+                    .withMetadata(allMetadata)
                     .build()
             );
         };
     }
 
+    /**
+     * See {@link #aboveThreshold(double, double, Map)}.
+     */
+    public static JobRuntimePredictionSelector aboveThreshold(Config config, Map<String, String> selectionMetadata) {
+        double runtimeThresholdInSeconds = config.getDouble("runtimeThresholdInSeconds", 600.0);
+        double sigmaThreshold = config.getDouble("sigmaThreshold", 60.0);
```

Comment:
default here seems to high, it should be probably 2, or 3 from experiments IIRC. @rostyboost confirm?

Reply:
depends what we consider "default". @tbak wanted to make it so that by default the threshold on `sigma` is a no-op, and we're only thresholding on P95. makes sense to me. in this case I'd argue for the default to be `Double.MAX_VALUE`

Reply:
`Double.MAX_VALUE` as no-op seems to be best here. Will do that